src/api.v2/controllers/qcController.js

```diff
@@ -0,0 +1,64 @@
+// const { OK } = require('../../utils/responses');
+const AWSXRay = require('aws-xray-sdk');
+
+const { createQCPipeline } = require('../helpers/pipeline/pipelineConstruct');
+const handleQCResponse = require('../helpers/pipeline/handleQCResponse');
+
+const getLogger = require('../../utils/getLogger');
+const parseSNSMessage = require('../../utils/parse-sns-message');
+
+const logger = getLogger('[QCController] - ');
+
+const runQC = async (req, res) => {
+  const { experimentId } = req.params;
+  const { processingConfig } = req.body;
+
+  logger.log(`Starting qc for experiment ${experimentId}`);
+
+  const { stateMachineArn, executionArn } = await createQCPipeline(
+    req.params.experimentId,
+    processingConfig || [],
+    req.headers.authorization,
+  );
+
+  logger.log(`Started qc for experiment ${experimentId} successfully, `);
+
+  res.json({ stateMachineArn, executionArn });
+};
+
+const handleResponse = async (req, res) => {
+  let result;
+
+  try {
+    result = await parseSNSMessage(req);
+  } catch (e) {
+    logger.error('Parsing initial SNS message failed:', e);
+    AWSXRay.getSegment().addError(e);
+    res.status(200).send('nok');
+    return;
+  }
+
+  const { io, parsedMessage } = result;
+
+  const isSnsNotification = parsedMessage !== undefined;
+  if (isSnsNotification) {
+    try {
+      await handleQCResponse(io, parsedMessage);
+    } catch (e) {
+      logger.error(
+        'pipeline response handler failed with error: ', e,
```

Comment:
for consistency with the log in gem2s handleResponse function, you can add `qc` before `pipeline` in the log message.